src/view/src/ContentView.c

```diff
@@ -462,19 +463,44 @@ STATUS contentViewAddItem(PContentView pContentView, UINT64 timestamp, UINT64 ac
         // Check if we need to evict based on the max buffer depth or temporal window
         CHK_STATUS(contentViewCheckAvailability(pContentView, &currentAvailability, &windowAvailability));
         if (!windowAvailability) {
-            // Move the tail first
-            pRollingView->tail++;
+            switch (pRollingView->bufferOverflowStrategy) {
+                case DROP_SINGLE_VIEW_ITEM:
+                    // Move the tail first
+                    pRollingView->tail++;
+
+                    // Callback if it's specified
+                    if (pRollingView->removeCallbackFunc != NULL) {
+                        // NOTE: The call is prompt - shouldn't block
+                        pRollingView->removeCallbackFunc(pContentView, pRollingView->customData, pTail, !currentAvailability);
+                    }
+                    break;
+
+                case DROP_UNTIL_FRAGMENT_START:
+                    do {
+                        // Move the tail first
+                        pRollingView->tail++;
+
+                        // Callback if it's specified
+                        if (pRollingView->removeCallbackFunc != NULL) {
+                            // NOTE: The call is prompt - shouldn't block
+                            pRollingView->removeCallbackFunc(pContentView, pRollingView->customData, pTail,
+                                                             !currentAvailability);
+                        }
+
+                        pTail = GET_VIEW_ITEM_FROM_INDEX(pRollingView, pRollingView->tail);
+                    } while (!CHECK_ITEM_FRAGMENT_START(pTail->flags) && pRollingView->tail != pRollingView->head);

```

Comment:
Could you double check whether you need to check CHECK_ITEM_STREAM_START or we always have both flags on the session start.

Reply:
yes we always have fragment start.